dali/operators/reader/nvdecoder/sequencewrapper.h

```diff
@@ -47,44 +47,28 @@ struct SequenceWrapper {
     timestamps.clear();
     timestamps.reserve(count);
 
-    int dev;
-    CUDA_CALL(cudaGetDevice(&dev));
-    std::unique_lock<std::mutex> lock{started_lock_};
-    if (started_) {
-      CUDA_CALL(cudaEventDestroy(event_));
+    if (!initialized_) {
+      CUDA_CALL(cudaEventCreateWithFlags(&event_, cudaEventBlockingSync | cudaEventDisableTiming));
     }
-    CUDA_CALL(cudaEventCreateWithFlags(&event_, cudaEventBlockingSync | cudaEventDisableTiming));
-    started_ = false;
+    initialized_ = true;
   }
 
   ~SequenceWrapper() {
-    std::unique_lock<std::mutex> lock{started_lock_};
-    if (started_) {
-      try {
-        CUDA_CALL(cudaEventDestroy(event_));
-      } catch (const std::exception &) {
-        // Something went wrong with releasing resources. We'd better die now.
-        std::terminate();
-      }
+    if (initialized_) {
+      CUDA_CALL(cudaEventDestroy(event_));
     }
   }
 
   void set_started(cudaStream_t stream) {
     CUDA_CALL(cudaEventRecord(event_, stream));
-    LOG_LINE << event_ << " recorded with stream " << stream << std::endl;
-    {
-      std::unique_lock<std::mutex> lock{started_lock_};
-      started_ = true;
-    }
-    started_cv_.notify_one();
   }
 
   void wait() const {
-    LOG_LINE << event_ << " wait to start" << std::endl;
-    wait_until_started_();
-    LOG_LINE << event_ << " waiting for sequence event" << std::endl;
-    CUDA_CALL(cudaEventSynchronize(event_));
-    LOG_LINE << event_ << " synchronized!" << std::endl;
+    if (initialized_) {
```

Comment:
```suggestion
    if (event_) {
```
However, is it no longer possible for `wait()` to be called before `set_started`? If not, this function will return immediately as if the event was signalled.

Reply:
I checked the flow and we don't need to call `wait` before `set_started`. Maybe that was needed for nvvl, where different thread scheduled the work and different waited for it.